tests/aggregation/tests.py

```diff
@@ -1163,3 +1163,20 @@ def test_group_by_exists_annotation(self):
             Exists(long_books_qs),
         ).annotate(total=Count('*'))
         self.assertEqual(dict(has_long_books_breakdown), {True: 2, False: 3})
+
+    def test_distinct_on_aggregate(self):
+        """
+        Supported aggregate using distinct=True should only use the distinct values.
+        """
+        authors = Author.objects.filter(book__in=[self.b5, self.b6])
+        self.assertEqual(authors.count(), 3)
+
+        # Selected author ages are 57 and 46
+        for aggregate, expected_result in (
+            (Avg, Approximate(51.5)),
+            (Count, Approximate(2)),
+            (Sum, Approximate(103))
+        ):
+            with self.subTest(aggregate=aggregate.__name__):
+                age_agg = authors.aggregate(age_agg=aggregate('age', distinct=True))
+                self.assertEqual(age_agg['age_agg'], expected_result)
```

Comment:
`Count(..., distinct=True)` is already tested in `test_count()` and it is not related with this patch, so we can refactor tests but in a separate commit, e.g.

- 1st commit: _"Moved test for distinct Count() to a separate test case."_:
```diff
diff --git a/tests/aggregation/tests.py b/tests/aggregation/tests.py
index ea11c02edc..2f667a0fe1 100644
--- a/tests/aggregation/tests.py
+++ b/tests/aggregation/tests.py
@@ -388,9 +388,6 @@ class AggregateTestCase(TestCase):
         vals = Book.objects.aggregate(Count("rating"))
         self.assertEqual(vals, {"rating__count": 6})
 
-        vals = Book.objects.aggregate(Count("rating", distinct=True))
-        self.assertEqual(vals, {"rating__count": 4})
-
     def test_count_star(self):
         with self.assertNumQueries(1) as ctx:
             Book.objects.aggregate(n=Count("*"))
@@ -403,6 +400,10 @@ class AggregateTestCase(TestCase):
         )
         self.assertEqual(aggs['distinct_ratings'], 4)
 
+    def test_distinct_on_aggregate(self):
+        books = Book.objects.aggregate(Count('rating', distinct=True))
+        self.assertEqual(books, {'rating__count': 4})
+
     def test_non_grouped_annotation_not_in_group_by(self):
         """
         An annotation not included in values() before an aggregate should be
```

- 2nd commit, fix and extra test cases (with `self.subTest`), e.g.
```python
    def test_distinct_on_aggregate(self):
        for aggregate, expected_result in (
            (Count, 4),
            (Sum, 16.5),
            (Avg, 4.125),
        ):
            with self.subTest(aggregate=aggregate):
                books = Book.objects.aggregate(ratings=aggregate('rating', distinct=True))
                self.assertEqual(books['ratings'], expected_result)
```

Also, I think that `Book.rating` would be better for testing `distinct` argument.